File: MutationMultiplicationTable/java/MutationMultiplicationTable.java
/**
 * 變種九九乘法表
 *
 * 輸出為1,m,2,m-1......
 *
 * 例如19*9輸出中的7
 *      7*1=1
 *      7*9=63
 *      7*2=14
 *
 *      ...
 *      依此類推
 */

import java.util.Scanner;

public class MutationMultiplicationTable{

	// 反向乘數來囉
	public static int reverseWay(int multiplicand, int round){
		int reverse = 9 - round + 1;
		System.out.println(multiplicand+" * "+reverse+" = "+multiplicand*reverse);
		return 0;
	}

	public static void main(String[] args){
		Scanner input = new Scanner(System.in);
		System.out.print("請輸入數字：");
		int go = input.nextInt();  //輸入要做多少*9的表
		for(int i=1; i<=go; i++){
			for(int j=1; j<=5; j++){
				System.out.println(i+" * "+j+" = "+i*j);
				if(j != 5){
					reverseWay(i, j);
				}
			}
			System.out.println();  // 被乘數+1前先換行比較好看
		}
	}
}

File: MutationMultiplicationTable/python/MutationMultiplicationTable.py
require = input('請輸入想要的被乘數字：')
for i in range(1, int(require)+1):  # 從1開始輸出到使用者指定的數字
	for j in range(1, 6):  # 5+1
		print(i, "*", j, "=", i*j)
		# 還沒到中間（數字5）之前都要準備倒轉乘數
		if j != 5:
			reverse = 9 - j + 1  # 看看現在乘數應該是多少
			print(i, '*', reverse, '=', i*reverse)
	print()  # 被乘數*9之後換一行空白再輸出比較好看
